lib/util.js

```diff
@@ -361,168 +371,150 @@ function formatValue(ctx, value, recurseTimes) {
-    if (typeof value === 'function') {
-      const ctorName = constructor ? constructor.name : 'Function';
-      return ctx.stylize(
-        `[${ctorName}${value.name ? `: ${value.name}` : ''}]`, 'special');
-    }
-    if (isRegExp(value)) {
-      return ctx.stylize(regExpToString.call(value), 'regexp');
-    }
-    if (isDate(value)) {
-      if (Number.isNaN(value.getTime())) {
-        return ctx.stylize(value.toString(), 'date');
-      } else {
-        return ctx.stylize(dateToISOString.call(value), 'date');
-      }
-    }
-    if (isError(value)) {
-      return formatError(value);
-    }
-    // now check the `raw` value to handle boxed primitives
-    if (typeof raw === 'string') {
-      formatted = formatPrimitiveNoColor(ctx, raw);
-      return ctx.stylize(`[String: ${formatted}]`, 'string');
-    }
-    if (typeof raw === 'symbol') {
-      formatted = formatPrimitiveNoColor(ctx, raw);
-      return ctx.stylize(`[Symbol: ${formatted}]`, 'symbol');
-    }
-    if (typeof raw === 'number') {
-      formatted = formatPrimitiveNoColor(ctx, raw);
-      return ctx.stylize(`[Number: ${formatted}]`, 'number');
-    }
-    if (typeof raw === 'boolean') {
-      formatted = formatPrimitiveNoColor(ctx, raw);
-      return ctx.stylize(`[Boolean: ${formatted}]`, 'boolean');
-    }
-    // Fast path for ArrayBuffer and SharedArrayBuffer.
-    // Can't do the same for DataView because it has a non-primitive
-    // .buffer property that we need to recurse for.
-    if (isAnyArrayBuffer(value)) {
-      return `${constructor.name}` +
-             ` { byteLength: ${formatNumber(ctx, value.byteLength)} }`;
-    }
-    if (isExternal(value)) {
-      return ctx.stylize('[External]', 'special');
-    }
-  }
+  const keyLength = keys.length + symbols.length;
+  const constructor = getConstructorOf(value);
+  const ctorName = constructor && constructor.name ?
+    `${constructor.name} ` : '';
 
   var base = '';
-  var empty = false;
   var formatter = formatObject;
-  var braces;
+  var braces = ['{', '}'];
 
   // We can't compare constructors for various objects using a comparison like
   // `constructor === Array` because the object could have come from a different
   // context and thus the constructor won't match. Instead we check the
   // constructor names (including those up the prototype chain where needed) to
   // determine object types.
   if (Array.isArray(value)) {
+    if (recurseTimes < 0)
+      return ctx.stylize('[Array]', 'special');
+
     // Unset the constructor to prevent "Array [...]" for ordinary arrays.
-    if (constructor && constructor.name === 'Array')
-      constructor = null;
-    braces = ['[', ']'];
-    empty = value.length === 0;
+    if (ctorName === 'Array ') {
+      braces = ['[', ']'];
+    } else {
+      braces = [`${ctorName}[`, ']'];
+    }
+    if (value.length === 0 && keyLength === 0)
+      return `${braces[0]}]`;
     formatter = formatArray;
+  } else if (isMapIterator(value)) {
```

Comment:
a long series of if - else if - else if statements can also be formatted as a `switch (true) { ... }` (note that switch does a strict comparison, ie matches not truthy but `true`)